validator/main.go

```diff
@@ -318,6 +320,27 @@ func main() {
 			Destination: &disableDevCharSymlinkCreation,
 			EnvVars:     []string{"DISABLE_DEV_CHAR_SYMLINK_CREATION"},
 		},
+		&cli.BoolFlag{
```

Comment:
I think we should avoid adding this env var to the validator container. I would much prefer if we had logic within the validator to introspect the owner of the pod via the kube api and then determine if the driver is gpu-operator managed or not. Adding an env var to function as a boolean flag does not seem right tome

Reply:
Ah, I just read your note. Looks like we are on the same page. Can we make the change in this PR instead of doing it in a follow-up?

Reply:
I agree, and I suggested this approach here: https://github.com/NVIDIA/gpu-operator/pull/747#discussion_r1633942404

Reply:
Done.